### PR TITLE
test: add restart driver e2e test

### LINKED ISSUE #154: add restart driver e2e test
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like in detail**

1. mount a pod
2. restart driver daemonset
3. make sure original pod could still access volume

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by dr0pdb**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:
This PR adds an e2e test to ensure that existing pods still have access to volume after restarting the driver daemonset.

**Which issue(s) this PR fixes**:

Fixes #154

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
none

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
sorry, I meant such scenario:
```
1. mount a pod
2. restart driver daemonset  (don't need to uninstall and reinstall whole driver)
3. make sure original pod could still access volume
```

**Comment by dr0pdb** [comment]
oh okay. will update this PR.

**Comment by dr0pdb** [comment]
@andyzhangx This is ready for review.

File: test/utils/restart_driver_daemonset.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    repo="./deploy"
+  fi
+fi
+
+if [ $ver != "master" ]; then
+	repo="$repo/$ver"
+fi
+
+echo "Deleting the Driver node daemonset, version: $ver ..."
+kubectl delete -f $repo/csi-azurefile-node.yaml --ignore-not-found
```
Comment (andyzhangx) [inline test/utils/restart_driver_daemonset.sh:37]: I think you could use current repo path, and don't use `ignore-not-found`
```
kubectl delete -f ./deploy/csi-azurefile-node.yaml
```

File: test/e2e/dynamic_provisioning_test.go [hunk 2]
```diff
@@ -704,6 +704,48 @@ var _ = ginkgo.Describe("Dynamic Provisioning", func() {
 		test.Run(cs, ns)
 	})
 
+	ginkgo.It("should create a volume after driver restart [kubernetes.io/azure-file] [file.csi.azure.com]", func() {
+		pod := testsuites.PodDetails{
```
[the remark this answers is not in the document]
Comment (dr0pdb) [reply test/e2e/dynamic_provisioning_test.go:713]: done

File: test/utils/restart_driver_daemonset.sh (hunk 1 above)
Comment (dr0pdb) [reply test/utils/restart_driver_daemonset.sh]: sure, done

**Comment by dr0pdb** [comment]
```bash
Error from server (NotFound): error when deleting "./deploy/csi-azurefile-node.yaml": daemonsets.apps "csi-azurefile-node" not found
```

@andyzhangx Any idea on why this is happening?

File: test/utils/restart_driver_daemonset.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+echo "Deleting the Driver node daemonset ..."
+kubectl delete -f ./deploy/csi-azurefile-node.yaml
```
Comment (andyzhangx) [inline test/utils/restart_driver_daemonset.sh:20]:
```console
kubectl delete -f ./deploy/csi-azurefile-node.yaml --ignore-not-found
kubectl delete -f ./deploy/csi-azurefile-node-windows.yaml --ignore-not-found
```

Comment (andyzhangx) [reply test/utils/restart_driver_daemonset.sh:20]: are you waiting for a few seconds after delete and then after install daemonset, e.g. `sleep 15`

Comment (dr0pdb) [reply test/utils/restart_driver_daemonset.sh]: Yes, I have added `sleep 15`. It's fixed now.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by andyzhangx** [comment]
can you also port this test to smb driver? thanks.

**Pull request merged**